modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/common/utils/dateFormatter.js

```diff
@@ -39,16 +44,92 @@ export const december = '12';
 export const arrayOfMonthsWith30Days = [3, 5, 8, 10];
 export const arrayOfMonthsWith31Days = [0, 2, 4, 6, 7, 9, 11];
 
+export const lastDayOfLastMonthDate = convertDateToString(
+	new Date(
+		new Date(new Date().setMonth(getCurrentMonth)).setDate(lastDayOfMonth)
+	)
+).split('-');
+
+export const firstDayOfLastMonthDate = convertDateToString(
+	new Date(new Date(new Date().setMonth(lastMonth)).setDate(fistDayOfMonth))
+).split('-');
+
 export const lastMonthDate = convertDateToString(
 	new Date(new Date().setMonth(lastMonth))
 ).split('-');
 
+export const lastDateOfLastYear = convertDateToString(
+	new Date(
+		new Date(
+			new Date(new Date().setFullYear(lastYear)).setMonth(
+				getCurrentMonth + 1
+			)
+		).setDate(lastDayOfMonth)
+	)
+).split('-');
+
 export const threeMonthsAgoDate = convertDateToString(
-	new Date(new Date().setMonth(threeMonthsAgo))
+	new Date(
+		new Date(new Date().setMonth(threeMonthsAgo)).setDate(fistDayOfMonth)
+	)
+).split('-');
+
+export const threeMonthsAgoDateLastYear = convertDateToString(
+	new Date(
+		new Date(
+			new Date(new Date().setFullYear(lastYear)).setMonth(threeMonthsAgo)
+		).setDate(fistDayOfMonth)
+	)
+).split('-');
+
+export const threeMonthsAgoDateLastYearLastDay = convertDateToString(
+	new Date(
+		new Date(
+			new Date(new Date().setFullYear(lastYear)).setMonth(
+				getCurrentMonth + 1
+			)
+		).setDate(lastDayOfMonth)
+	)
+).split('-');
+
+export const sixMonthsAgoDateLastYearLastDay = convertDateToString(
+	new Date(
+		new Date(
+			new Date(new Date().setFullYear(lastYear)).setMonth(
+				getCurrentMonth + 1
+			)
+		).setDate(lastDayOfMonth)
+	)
 ).split('-');
 
 export const sixMonthsAgoDate = convertDateToString(
-	new Date(new Date().setMonth(sixMonthsAgo))
+	new Date(
+		new Date(new Date().setMonth(sixMonthsAgo)).setDate(fistDayOfMonth)
+	)
+).split('-');
+
+export const sixMonthsAgoDateLastYear = convertDateToString(
+	new Date(
+		new Date(
+			new Date(new Date().setFullYear(lastYear)).setMonth(sixMonthsAgo)
+		).setDate(fistDayOfMonth)
+	)
+).split('-');
```

Comment:
Try to do it in a generic way, like this:

```suggestion

const generateDateThroughPeriod = (year, period) =>{
     const generatedDate = convertDateToString(
	new Date(
		new Date(
			new Date(new Date().setFullYear(year)).setMonth(period)
		).setDate(fistDayOfMonth)
	)
).split('-');

   return generatedDate;
}

export const sixMonthsAgoDateLastYear = generateDateThroughPeriod(lastYear,sixMonthsAgo)
```

Reply:
Done.